### PR TITLE
Roles for GitHub actions

### PR DESCRIPTION
**Pull request by MancunianSam**
- Add roles on the management account.
- Add policy to allow release job to upload to backend code bucket
- Add roles for github actions

### CODE REVIEW

File: modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl [hunk 1]
```diff
@@ -53,6 +47,7 @@
         "arn:aws:iam::${account_id}:policy/TDRSNSPublish${title(environment)}",
         "arn:aws:iam::${account_id}:policy/TDRVpcFlowlogPolicy${title(environment)}",
         "arn:aws:iam::${account_id}:policy/TDRYaraAvPolicy",
+        "arn:aws:iam::${account_id}:policy/UpdateWAFAndSecurityGroupsPolicy${title(environment)}",
```
Comment (TomJKing) [inline modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl:50]: Rename with `TDR` prefix?

File: root.tf [hunk 2]
```diff
@@ -361,3 +370,52 @@ module "periodic_ecr_image_scan_event" {
   rule_name               = "ecr-scan"
   lambda_event_target_arn = module.periodic_ecr_image_scan_lambda.ecr_scan_lambda_arn
 }
+
+module "run_e2e_tests_role" {
+  source             = "./tdr-terraform-modules/iam_role"
+  assume_role_policy = templatefile("${path.module}/templates/iam_role/github_assume_role.json.tpl", { account_id = data.aws_ssm_parameter.mgmt_account_number.value, repo_name = "tdr-e2e-tests:*" })
+  common_tags        = local.common_tags
+  name               = "GithubActionsE2ETestsMgmt"
```
Comment (TomJKing) [inline root.tf:378]: Maybe add `TDR` prefix to name?

File: templates/iam_policy/github_code_bucket.json.tpl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Action": ["s3:GetObject","s3:PutObject", "s3:ListBucket"],
```
Comment (TomJKing) [inline templates/iam_policy/github_code_bucket.json.tpl:7]: Maybe order alphabetically?

File: templates/iam_policy/github_sbt_dependencies_policy.json.tpl [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "Statement": [
+    {
+      "Action": [
```
Comment (TomJKing) [inline templates/iam_policy/github_sbt_dependencies_policy.json.tpl:4]: Maybe order alphabetically?

File: templates/iam_policy/github_sbt_dependencies_policy.json.tpl [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      "Resource": [
+        "arn:aws:s3:::tdr-snapshots-mgmt/*",
+        "arn:aws:s3:::tdr-snapshots-mgmt",
+        "arn:aws:s3:::tdr-releases-mgmt/*",
+        "arn:aws:s3:::tdr-releases-mgmt"
+      ],
+      "Sid": ""
+    }
+  ],
+  "Version": "2012-10-17"
```
Comment (TomJKing) [inline templates/iam_policy/github_sbt_dependencies_policy.json.tpl:19]: Maybe put the `Version` field at the beginning to make consistent with other policy declarations?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments

File: modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply modules/environment-roles/templates/shared_iam_terraform_policy.json.tpl:50]: I've done this one and the role as well and moved them so they're alphabetical.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**